src/main/scala/com/fulcrumgenomics/illumina/SampleSheet.scala

```diff
@@ -98,27 +98,28 @@ object SampleSheet {
       case None => throw new IllegalArgumentException("Could not find the header for sample data.")
     }).map(_.toUpperCase.trim)
 
-    // get the rows (values), so skip "[Data]" and the header
+    // get the rows (values), so skip "[Data]" and header line
     val lineNumber = preData.size + 2 // 0-based
-    val maybeSampleDatums = postData.drop(2).zipWithIndex.map { case (line, rowNumber) =>
-        val values = line.split(SplitRegex, -1)
-        // treat as empty line if all columns are empty (empty line or comma-only line, often added by spreadsheet editors)
-        if (values.forall(_.trim.isEmpty)) None
-        // check we have the correct # of columns
-        else if (values.size != header.length) {
-          throw new IllegalArgumentException(s"Found a line with a mismatching number of columns at sample data row #${rowNumber + 1}: " + line)
-        } else {
-          Some(header.zip(values.map(_.trim)).toMap)
+    val sampleDatums = postData
+      .drop(2) // skip the "[Data]" and header line
+      .zipWithIndex // add the row #
+      .map { case (line, rowNumber) => (line.split(SplitRegex, -1), rowNumber) } // just split the values
+      .reverse // to skip empty lines, and lines with all empty values, at the end of the sample
+      .dropWhile { case (values, rowNumber) => values.isEmpty || values.forall(_.trim.isEmpty) } // no values, regardless of number, so drop/skip it
+      .reverse // put back in the correct order
+      .map { case (values, rowNumber) =>
+        if (values.isEmpty || values.forall(_.trim.isEmpty)) { // empty line, or all values are empty
+          val emptyRows = values.zipWithIndex.filter(_._1.isEmpty).map(_._2).map(_ + 1) // 1 indexed
+          throw new IllegalArgumentException(s"Found empty rows in sample data section: ${emptyRows.mkString(", ")}")
```

Comment:
`emptyRows` looks like it is really the empty columns for a single empty row. 

```suggestion
          throw new IllegalArgumentException(s"Found empty row in sample data section: ${rowNumber}")
```